src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/CompositeModelBinder.cs

```diff
@@ -36,24 +37,26 @@ public CompositeModelBinder(IList<IModelBinder> modelBinders)
         /// <inheritdoc />
         public IList<IModelBinder> ModelBinders { get; }
 
-        public virtual Task<ModelBindingResult> BindModelAsync(ModelBindingContext bindingContext)
+        public virtual async Task BindModelAsync(ModelBindingContext bindingContext)
```

Comment:
By making this async, we've undone an optimization here. Notice `BindModelAync` now has a state machine. If there's not an easy way to address this, let's not block on it.

*Before*:
![image](https://cloud.githubusercontent.com/assets/1430011/12689551/822b0314-c692-11e5-8213-67ee17bc0333.png)

*After:*
![image](https://cloud.githubusercontent.com/assets/1430011/12689560/9117bdfe-c692-11e5-8b6f-f0bc8c159119.png)